src/OpenSearch.Net.Auth.AwsSigV4/AwsSigV4HttpConnection.cs

```diff
@@ -0,0 +1,65 @@
+/* SPDX-License-Identifier: Apache-2.0
+*
+* The OpenSearch Contributors require contributions made to
+* this file be licensed under the Apache-2.0 license or a
+* compatible open source license.
+*
+* Modifications Copyright OpenSearch Contributors. See
+* GitHub history for details.
+*
+*  Licensed to Elasticsearch B.V. under one or more contributor
```

Comment:
For new files you do not need to retain anything other than the SPDX header. We don't need to license this code explicitly to anyone since it's APLv2. See https://github.com/opensearch-project/.github/blob/main/CONTRIBUTING.md#license-headers.

Reply:
The reason for them being that exact header, is there's a [workflow](https://github.com/opensearch-project/opensearch-net/blob/main/.github/workflows/license.yml) that runs this [script](https://github.com/opensearch-project/opensearch-net/blob/main/.github/check-license-headers.sh) to assert all `.cs` files have this [header](https://github.com/opensearch-project/opensearch-net/blob/main/.github/license-header.txt).

Additionally it's configured to be auto-added by IDEs [here](https://github.com/opensearch-project/opensearch-net/blob/main/OpenSearch.sln.DotSettings#L17-L43).

I'm more than happy to fix the license headers, but we need a decision on what to do with the workflow, I assume just expand it to check for either the "legacy" license or just the SPDX header?

Reply:
I see. You can do this separately/before/after this PR - thank you. The workflow should check only SPDX, and we need to pay attention to not removing any headers from old(er) files in PRs.

Reply:
I created a separate issue to discuss the desired approach here: https://github.com/opensearch-project/opensearch-net/issues/69